### ISSUE TITLE
Associations Type Clarification

### ISSUE THREAD

**Issue by brandonopened** [opensalt/opensalt#675]
As an editor, I want to see a short description of each type of association type directly on the Make Association screen: 

+-------------+-------------------------------------------------------+
|Association    | Short Description                                                               |
+-------------+------------------------------------------------------ +
|IsRelatedTo    |                                                                                              |
+-------------+-------------------------------------------------------+

**Comment by brandonopened** [comment]
I see that the make association dialog box here: https://github.com/opensalt/opensalt/blob/44605a8400b2cfd1de5ceef2c186ab042650f3bb/templates/framework/ls_association/_new_association_component.html.twig

calls for "associationFormType". and later I see 'assocTypes'... which makes me think adding alternative labels here would be harder, because those types are hard-wired in a list somewhere that dialog box is simply displaying. 

But for general usability I think we should try. @jmarks I don't really like arbitrary descriptions above but we also should think about how most general users don't want to go look up the implementation guide or information model.

**Comment by jmarks** [comment]
Regarding labels, they should be consistent if possible.

Regarding the descriptions, maybe we could link to the best practices/implementation guide (At some point) for more detail and examples?

**Comment by brandonopened** [comment]
I agree the descriptions should have a link to the implementation guide. But for OpenSALT usability... something shorter still?
